rust/connlib/libs/tunnel/src/tun_darwin.rs

```diff
@@ -150,19 +153,22 @@ impl IfaceDevice {
             }
 
             if addr.sc_id == info.ctl_id {
-                return Ok(Self { fd });
+                callbacks.on_set_interface_config(config.ipv4, config.ipv6, DNS_SENTINEL);
```

Comment:
I think we want to move this call to the top of this function and update the Swift side to behave like the Android side:

1. Call Session.connect
2. Create the tunnel interface in the `onSetInterfaceConfig` callback

This will ensure the `fd` is opened for us to find it here.

Reply:
cc @roop ^

Reply:
In Apple platforms, the tunnel interface / fd is created by the OS in the tunnel process before `startTunnel()` (therefore, before we call `connlib.connect()`), so the Rust code *can* find the `fd` after `onSetInterfaceConfig` returns.

The Swift code applies the network settings and brings up the tunnel only at `onTunnelReady`.

Reply:
I see. In that case it won't matter too much whether we call `on_set_interface_config` here or up above.

I thought it might be cleaner on the Rust side for connlib to always expect a `RawFd` to be returned from `on_set_interface_config` callback, and if it's `-1`, continue with the platform-specific approach to obtaining an actual `fd`.